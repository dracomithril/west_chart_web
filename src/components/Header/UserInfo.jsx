/**
 * Created by Gryzli on 07.04.2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Button, Image, Well } from 'react-bootstrap';
// TODO save information about user to database, do it somewhere else
// const { writeUserData } = require('./../../utils');

const UserInfo = ({ fb_user, sp_user, onLogoutClick }) => (
  <Well bsClass="logged">
    <div>
      <Image src={fb_user.picture_url} circle />
      <span>{`Hi, ${fb_user.first_name}`}</span>
      <br />
      <span>it is nice to see you again.</span>
      <br />
      {sp_user.id !== undefined && (
        <div className="spotify">
          <i className="fa fa-spotify" />
          <span>{sp_user.id}</span>
        </div>
      )}
      <div style={{ textAlign: 'center' }}>
        <Button
          bsStyle="warning"
          onClick={() => {
            onLogoutClick();
          }}
        >
          Sign out
        </Button>
      </div>
    </div>
  </Well>
);

UserInfo.propTypes = {
  fb_user: PropTypes.object,
  sp_user: PropTypes.object,
  onLogoutClick: PropTypes.func,
};

export default UserInfo;
