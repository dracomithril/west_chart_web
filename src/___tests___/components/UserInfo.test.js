/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import UserInfo from '../../components/Header/UserInfo';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() });
const initial_state = require('../data/initial_state.json');

const mockStore = configureMockStore([]);

describe('<UserInfo/>', () => {
  it('renders without crashing', () => {

    const store = mockStore(initial_state);
    const wrapper = shallow(
      <UserInfo fb_user={initial_state.user} sp_user={initial_state.sp_user}/>, {
        context: { store },
        childContextTypes: { store: PropTypes.object }
      }
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
