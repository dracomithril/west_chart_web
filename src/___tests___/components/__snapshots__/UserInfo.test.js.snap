// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo/> renders without crashing 1`] = `
<Well
  bsClass="logged"
>
  <div>
    <Image
      bsClass="img"
      circle={true}
      responsive={false}
      rounded={false}
      thumbnail={false}
    />
    <span>
      Hi, undefined
    </span>
    <br />
    <span>
      it is nice to see you again.
    </span>
    <br />
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="warning"
        disabled={false}
        onClick={[Function]}
      >
        Sign out
      </Button>
    </div>
  </div>
</Well>
`;
